File: big_data_learning/spark-scala-java-learning/src/main/java/com/shawn/study/big/data/spark/java/core/rdd/basic/GlomDemo.java
package com.shawn.study.big.data.spark.java.core.rdd.basic;

import java.util.Arrays;
import java.util.List;
import org.apache.commons.lang.StringUtils;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class GlomDemo {

  public static void main(String[] args) {
    SparkConf conf = new SparkConf().setAppName("glom-demo").setMaster("local[*]");
    JavaSparkContext context = new JavaSparkContext(conf);
    JavaRDD<Integer> javaRDD = context.parallelize(Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8), 2);
    List<List<Integer>> list = javaRDD.glom().collect();
    System.out.println(StringUtils.join(list, ", "));
    context.close();
  }
}

File: big_data_learning/spark-learning/com/shawn/study/spark/core/api/glom-demo.py
from pyspark import SparkContext, SparkConf

if __name__ == '__main__':
    conf = SparkConf().setAppName(value="glom-demo").setMaster(value="local[*]")
    context = SparkContext(conf=conf)
    rdd = context.parallelize(c=[1, 2, 3, 4, 5, 6, 7, 8], numSlices=3)
    print(rdd.glom().collect())
    context.stop()
